bsip-0022.md

```diff
@@ -9,113 +9,439 @@
+
+The behavior of that functionality is consistent with the intent of the proposal in that any changes to an account voting slate, even the removal of proxy delegation, is considered an updated an acccount's last voting time.
+
+
+### Calculation of Voting Power
+
+The calculation of voting power is currently performed by the [vote_tally_helper](https://github.com/bitshares/bitshares-core/blob/master/libraries/chain/db_maint.cpp#L1178) during chain maintenance.
 
-## Should the committee have the power to blacklist users from voting?
+It is here that the vote decay could be applied by calculating a voting stake multiplier (&alpha;) which for vote decay would be equal to or less than one (&alpha; &le; 1).  This new function (`calculate_voting_power(opinion_account)`) will calculate the multipler as a function of the current decay parameters and the account's age of last vote.
 
-Whilst exchanges/services voting with user funds (without permission) is bad, such functionality opens the doors for normal users to be included in the blacklist. One would hope that the network would vote such a committee out, but what's to stop the committee then blacklisting users that don't vote for them (or who vote for competitor committee users) so as to maintain power indefinetley? Currently highly unlikely, but still theoretically plausible.
+Any account may have four (4) different multipliers for each of the three (3) referendum categories and the one (1) proxy delegation.  Therefore the `calculate_voting_power(opinion_account)` function can return a multiple-valued structure:
+
+```
+struct voting_power {
+   double for_witness;
+   double for_committee;
+   double for_worker;
+   double for_proxy_delegation;
+};
+```
+
+The developer is free to to code the `calculate_voting_power()` function in an efficient way to avoid potential performance problems with this calculation.
+
+An [implementation of vote decay](https://github.com/peerplays-network/peerplays/blob/qa_gpos_18.04/libraries/chain/db_maint.cpp#L728-L785) that is [proposed for Peerplays](#peerplays-vote-decay) may be helpful for comparison.
+
+This multiplier may then be used inside of the account loop (https://github.com/bitshares/bitshares-core/blob/master/libraries/chain/db_maint.cpp#L1182) by checking what is being voted for by the account and then decaying with the appropriate multiplier.
+
+```
+for( vote_id_type id : opinion_account.options.votes )
+    ...
+    if(vote_type == witness)
+       d._vote_tally_buffer[offset] += voting_power.witness * voting_stake;
+...
+```
+
+The subsequent logic of `db_maint` will remain unchanged.
+
+
+### <div id="software-specs-api"/> API Calls
+
+- An API call shall be added, or an existing one modified, to get the blockchain's current vote decay parameters
+
+- An API call shall be added, or an existing one modified, to get an account holder's date of last vote
```

Comment:
This should already exist from https://github.com/bitshares/bitshares-core/pull/1449/